### PR TITLE
Add a default 6-minute timeout on CD

### PR DESCRIPTION
**Pull request by ksigmund**

### CODE REVIEW

**Comment by ksigmund** [comment]
Addresses #105

**Comment by ByAgenT** [comment]
How did we decide on 6 minutes? Are we sure we don't have customers who succeeding scanning but for whom 6 minutes is not enough? Maybe we could make it configurable, for example with environment variable?

File: src/Microsoft.Sbom.Api/Utils/ComponentDetectorCachedExecutor.cs [hunk 1]
```diff
@@ -18,6 +19,7 @@ public class ComponentDetectorCachedExecutor
         private readonly ILogger log;
         private readonly ComponentDetector detector;
         private ConcurrentDictionary<int, ScanResult> results;
+        private readonly TimeSpan componentDetectionTimeout = TimeSpan.FromMinutes(6);
```
Comment (edgarrs) [inline src/Microsoft.Sbom.Api/Utils/ComponentDetectorCachedExecutor.cs:22]: Should we consider this to be overridable using a parameter or env var?

Comment (edgarrs) [reply src/Microsoft.Sbom.Api/Utils/ComponentDetectorCachedExecutor.cs:22]: If not, let's bump the timeout to make sure we allocate enough time for large codebases

Comment (ksigmund) [reply src/Microsoft.Sbom.Api/Utils/ComponentDetectorCachedExecutor.cs:22]: I would prefer making this configurable as well.  Next time let's make that an explicit requirement in the task ;-)

**Comment by ksigmund** [comment]
> How did we decide on 6 minutes? Are we sure we don't have customers who succeeding scanning but for whom 6 minutes is not enough? Maybe we could make it configurable, for example with environment variable?

Per offline discussion, this will be overridable via the `AdditionalComponentDetectorArgs` parameter and will default to 15 minutes.

File: src/Microsoft.Sbom.Api/Utils/ComponentDetectionCliArgumentBuilder.cs [hunk 2]
```diff
@@ -17,13 +17,18 @@ public class ComponentDetectionCliArgumentBuilder
         private string action;
         private VerbosityMode verbosity = VerbosityMode.Quiet;
         private string sourceDirectory;
-        private Dictionary<string, string> detectorArgs = new Dictionary<string, string>();
+        private Dictionary<string, string> detectorArgs = new Dictionary<string, string>() { 
+            { TimeoutArgsParamName, TimeoutDefaultSeconds.ToString() }
```
Comment (edgarrs) [inline src/Microsoft.Sbom.Api/Utils/ComponentDetectionCliArgumentBuilder.cs:21]: How does a user override this timeout?

Comment (ksigmund) [reply src/Microsoft.Sbom.Api/Utils/ComponentDetectionCliArgumentBuilder.cs:21]: Users can override this by adding "Timeout={timeout in seconds}" to the list of `AdditionalComponentDetectorArgs`.

**Comment by ksigmund** [comment]
> How did we decide on 6 minutes? Are we sure we don't have customers who succeeding scanning but for whom 6 minutes is not enough? Maybe we could make it configurable, for example with environment variable?

Apparently this was a WAG.  For now we'll default to 15 minutes and allow the user to override it.

**Review by aasim — APPROVED** [review]

**Pull request merged**